pygls/protocol.py

```diff
@@ -509,13 +509,12 @@ def bf_initialize(self, params: InitializeParams):
 
         client_capabilities = params.capabilities
         root_uri = params.rootUri
-        root_path = params.rootPath
-        workspace_folders = params.workspaceFolders or []
-
-        if root_uri is None:
-            root_uri = from_fs_path(root_path) if root_path is not None else ''
-
-        self.workspace = Workspace(root_uri, self)
+        root_path = getattr(params, 'rootPath', None)
+        workspace_folders = getattr(params, 'workspaceFolders', [])
+        client_capabilities = getattr(params, 'capabilities', None)
```

Comment:
This line should be removed (line 510. is fine).

Reply:
Oops, sorry, big mistake there.

Reply:
By removing line 514, line 510 should raise an attribute error if `capabilities` is `None`.

Reply:
Attribute error will be raised if request object does not have capabilities field.

@yorodm Good observation. This is from LSP specs:
```
/**
    * The rootUri of the workspace. Is null if no
    * folder is open. If both `rootPath` and `rootUri` are set
    * `rootUri` wins.
*/
```
So, _rootUri_ _can_ be _None_, but it's required, not sure if I got this.

I would say, just remove line 514, and if it's working on your end, we should be fine for now. The goal of this PR is to fix the bug that you detected, so you can move on with your project that is actually using `pygls`.